### PR TITLE
Free tickets

### PR DESCRIPTION
**Pull request by kirknorthrop**
Please also review tests and ensure all cases are covered?

### CODE REVIEW

File: tickets/models.py [hunk 1]
```diff
@@ -118,6 +137,20 @@ def invitation(self):
...
+                setattr(self, day, (day in days))
+            self.save()
+
+    @property
+    def is_free_ticket(self):
+        return self.free_reason is not None
+
+    @property
+    def is_changeable(self):
+        return self.free_reason in CHANGEABLE_REASONS
```
[the remark this answers is not in the document]
Comment (kirknorthrop) [reply tickets/models.py]: We have allowed some sponsors this year to use their tickets “creatively” by having one ticket with one person coming sat/sun and another mon/tue. Obviously we do not want these tickets to be editable.

[the remark this answers is not in the document]
Comment (inglesp) [reply tickets/models.py]: Understood.  You can always make sponsor tickets editable later on if it turns out you get people changing their mind about which three days they're coming.

[the remark this answers is not in the document]
Comment (kirknorthrop) [reply tickets/models.py]: Yes. I just figured that it was sensible to have that restriction - the only reason I've only enabled it for Django Girls is that's the only set of free tickets I have to create so far!

File: tickets/tests/test_views.py [hunk 2]
```diff
@@ -444,6 +444,114 @@ def test_when_not_authorized(self):
...
+        # Arrange
+        bob = factories.create_user('Bob')
+        ticket = factories.create_completed_free_ticket(bob)
+        form_data = {
+            'days': ['sun', 'mon', 'tue'],
+        }
+        self.client.force_login(ticket.owner)
+
+        # Act
+        self.client.post(f'/tickets/tickets/{ticket.ticket_id}/', form_data, follow=True)
```
[the remark this answers is not in the document]
Comment (kirknorthrop) [reply tickets/tests/test_views.py]: No - on the basis that if people don't know there isn't a POST endpoint there, we shouldn't reward them for trying to hack our system with an error message.

**Pull request merged**